src/background/openURLs.js

```diff
@@ -53,12 +52,21 @@ openURLs.getFilter = identifiers => {
 };
 
 openURLs.handleRequest = details => {
-    const refresh = details.url.indexOf(openURLs.identifiers.refreshTab) >= 0;
-    const url = details.url.replace(`${openURLs.identifiers[refresh ? "refreshTab" : "openTab"]}/`, "");
+    const url = new URL(details.url);
+    const refresh = url.hostname.indexOf(openURLs.identifiers.refreshTab) >= 0;
+    // URL will look like: "http://opensametab.morphic.org/redirect/https%3A%2F%2Fexample.com%2F"
+    const destination = url.pathname.replace(/^\/(redirect\/)?(.*)/, (match, c1, c2) => decodeURIComponent(c2));
 
-    openURLs.openTab(url, refresh);
+    let success;
+    try {
+        const destinationUrl = new URL(destination);
+        openURLs.openTab(destinationUrl, refresh);
+        success = true;
+    } catch (e) {
+        success = false;
+    }
```

Comment:
It appears that this try/catch block is to guard against malformed URLs used for the destinationUrl. e.g. `refresh/https%3A%2F%2Fexample.com%2F`. However, `openURLs` is an `async` function. So I don't believe that it will wait to make sure that it doesn't throw an error. You could use `await` on openURLs but will need to convert openURLs.handleRequest into an `async` function as well. According to [webRequest.onBeforeRequest](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/onBeforeRequest) the handler can return a promise that resolves with a BlockingResponse. You'll also have to test that the polyfill can handle this feature for use in Chrome.

Reply:
YEs. In fact, that's why I moved the url parsing out of `openURLs`.

However, I've now moved the call outside of the try block to make my intention clearer.